ext/curl/interface.c

```diff
@@ -3076,6 +3179,42 @@ PHP_FUNCTION(curl_getinfo)
 		if (curl_easy_getinfo(ch->cp, CURLINFO_LOCAL_PORT, &l_code) == CURLE_OK) {
 			CAAL("local_port", l_code);
 		}
+#endif
+#if LIBCURL_VERSION_NUM >= 0x073200 /* Available since 7.50.0 */
+		if (curl_easy_getinfo(ch->cp, CURLINFO_HTTP_VERSION, &l_code) == CURLE_OK) {
+			CAAL("http_version", l_code);
+		}
+#endif
+#if LIBCURL_VERSION_NUM >= 0x073400 /* Available since 7.52.0 */
+		if (curl_easy_getinfo(ch->cp, CURLINFO_PROTOCOL, &l_code) == CURLE_OK) {
+			CAAL("protocol", l_code);
+		}
+		if (curl_easy_getinfo(ch->cp, CURLINFO_PROXY_SSL_VERIFYRESULT, &l_code) == CURLE_OK) {
+			CAAL("ssl_verifyresult", l_code);
+		}
+		if (curl_easy_getinfo(ch->cp, CURLINFO_SCHEME, &s_code) == CURLE_OK) {
+			CAAS("scheme", s_code);
+		}
+#endif
+#if LIBCURL_VERSION_NUM >= 0x073700 /* Available since 7.55.0 */
+		if (curl_easy_getinfo(ch->cp, CURLINFO_CONTENT_LENGTH_DOWNLOAD_T, &co) == CURLE_OK) {
+			CAAL("content_length_download", (long) co);
```

Comment:
Hi Pierrick, long time! Are these (long) casts needed here, as the CAAL macro does cast to (zend_long) from the curl_off_t, there shouldn't be any data loss here

Reply:
Assuming `curl_off_t` is similar to POSIX `off_t`, the intermediate cast to `long` would be bad for 64bit Windows.

Reply:
I'm going to remove the (long) cast.